### DISCUSSION TITLE
There is an error message: 'glyphsUrl is not set'

### DISCUSSION THREAD

**Discussion by mdxiaohu** [maplibre/maplibre-gl-js#2760] [Q&A]
I use geojson point data .
Set the layer style to symbol .
Properties in the geojson attribute .
The values of certain internal attributes are numbers or strings .
for example : 

```
properties: {
    coordinate: "wgs84"
    name: "日本"
    num: 123456
}

text-field: ['get', 'name']

or

text-field: ['get', 'num']

```
When dynamically switching to 'text-field' .
There are hundreds of points in the data .
Sometimes the page will report an error .

`Error {message: 'glyphsUrl is not set'}`

I guess it's because some glyphs don't support it .
Is there any good way .
Excuse me . @HarelM

**Comment by mdxiaohu** [comment]
I can be certain that this should be a font issue .
I execute the following code when creating a map .

```
map.setGlyphs('https://demotiles.maplibre.org/font/{fontstack}/{range}.pbf');
map.setSprite('https://api.maptiler.com/maps/positron/sprite');
```
Now the map can display text .
Alphanumerics and symbols can be displayed normally, but not in Chinese .
I don't know why, it doesn't support Chinese. I may have to continue searching for relevant examples .
I don't know if I can customize the font file to 'Microsoft Yahei Font' .

**Comment by wipfli** [comment]
I think you always need to specify a glyph url if you have `text-field` somewhere in your style. But keep in mind that Chinese symbols (and Japanese, and Korean) are independent of the font file you specify. It uses a HTML canvas to render these symbols and the default font of your browser.

**Comment by mdxiaohu** [comment]
I'm glad to receive a reply .
I have reviewed the official document and performed the following actions .
```
map.setGlyphs('https://demotiles.maplibre.org/font/{fontstack}/{range}.pbf');
map.setSprite('https://api.maptiler.com/maps/positron/sprite');
```
Numbers and letters can be displayed normally, but not in Chinese .
I haven't found the answer yet .

**Comment by mdxiaohu** [comment]
I should provide more examples to help identify the root cause of this problem .
Unfortunately, the browser did not report any errors .
This is an example of the official website I am currently using .

`https://codepen.io/pen?&prefill_data_id=c1c718e9-59cb-49f3-a4a2-c35f04ffbfdc`

![Snipaste_2023-07-03_22-09-58](https://github.com/maplibre/maplibre-gl-js/assets/42248614/687cfdb3-5f0a-432e-b687-2568aac2742d)
This is its map style .

**Comment by mdxiaohu** [comment]
This is the style of my current map .
![Snipaste_2023-07-03_22-14-27](https://github.com/maplibre/maplibre-gl-js/assets/42248614/5a353996-9bd6-4041-bd13-f7d7647d0c33)
I installed fonts for the window computer .
for example: 

` text-font: ["Open Sans Semibold", "Arial Unicode MS Bold"]`

It seems to have no effect .
Chinese still cannot be displayed .
I always think this is a font issue, but there may also be other issues .

**Reply by wipfli** [reply]
Please have a look at https://maplibre.org/maplibre-gl-js-docs/example/local-ideographs
